public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -148,7 +148,11 @@ export function quoteWithEscape(input: string) {
 /**
  * Quotes string only when reserved characters are used
  */
+
+// The list of reserved characters that indicate we should be escaping the label key / value are
+// { } ! = ~ , \ " ' ` and any whitespace (\s), encoded in the regular expression below
```

Comment:
You can link to this PR which has the grammar written out https://github.com/prometheus/alertmanager/pull/3453.

Reply:
Good idea! Let me append that